Feature: Documentation Translation (#250)
simde-utc/portail: documentation/frontend/redux.md

Redux is used in order to manage data in a unique store.
All Redux's implementation is done in the `redux/` folder.

For more information on Redux, [have a look at this playlist](https://www.youtube.com/watch?v=1w-oQ-i1XB8&index=15&list=PLoYCgNOIyGABj2GQSlDRjgvXtqfDxKm5b).

## The 3 parts of Redux

### Action types

First, we have to define **action types**. They are constant strings which unable to identify actions to perform in the different reducers. It is recommanded to define them in the `types.js` file. In order to define CRUD actions sets more easily, use [`createCrudTypes`](#Action-types-creaters).

### Actions

Actions are objects that are scattered in the reducers. they generally content in a `type` and in a `payload`. They are defined in the `actions.js` file.

### Reducers

Reducers are fonctions with the following prototype: `function(prevState, action)`.

From the current status of the store `prevState` and of the dispatched action `action`, they return a new state (possibly `prevState` if the reducer doesn't have to do any modifications).

Do not directly modify `prevState`, make a copy of it. They are defined in the `reducers.js` file.

## CRUD creaters

> As th portal's api mainly follows CRUD (Create Read Update Delete) design, most part of the Redux actions also follows it. That is why we have created creaters of action types, of actions and of CRUD reducers. These functions are in `react/utils.js`.

Reviewer: ```suggestion
As the portal's api mainly follows CRUD (Create Read Update Delete) design, most part of the Redux actions also follow it. That is why we have created creators of action types, of actions and of CRUD reducers. These functions are in `react/utils.js`.
```

Ressources names are put in the singular by convention.

> ### Action types creaters

Reviewer: ```suggestion
### Action types creators
```

`createCrudTypes(name)` creates a set of types that follows the following scheme from a resources name `name` : 

This set of types is used by the following functions.

### Creater of actions

`createCrudActionSet(actionTypes, uri, overrides = {})` creates a set of CRUD actions from these parameters :
- `actionTypes` : a set of actions generated from the `createCrudTypes` function.

> - `uri` : the url piece located after the domain name. It unables to access the right resource in the API. For example `assos` unables to access to `https://assos.utc.fr/api/v1/assos`

Reviewer: ```suggestion
- `uri` : the url piece located after the domain name. It enables to access the right resource in the API. For example `assos` enables to access to `https://assos.utc.fr/api/v1/assos`
```

- `overrides` : an object unabling to replace and to add actions to the action set.

### Creater of reducer

`createCrudReducer(actionTypes, initialState = initialCrudState, overrides = {})` creates a CRUD reducer from the following parameters :
- `actionTypes` : an actions set generated from the `createCrudTypes` function
- `initialState` the initial state by default `initialCrudState`, can be replaced by extension of this one

> - `overrides` : an object unabling to replace and to add actions gestions to the reducer.

Reviewer: ```suggestion
- `overrides` : an object that enable to replace and to add actions managing to the reducer.
```

const articleReducer = createCrudReducer(articleTypes, customInitialState, overrides)
```

> which corresponds to (in a simplified way) à :

Reviewer: ```suggestion
which corresponds to (in a simplified way) :
```
